torchgeo/datasets/geo.py

```diff
@@ -852,8 +852,10 @@ def _merge_dataset_indices(self) -> None:
             for hit2 in ds2.index.intersection(hit1.bounds, objects=True):
                 box1 = BoundingBox(*hit1.bounds)
                 box2 = BoundingBox(*hit2.bounds)
-                self.index.insert(i, tuple(box1 & box2))
-                i += 1
+                new_box = box1 & box2
+                if new_box.area > 0:
```

Comment:
I'm on the fence about this change. We also support point datasets like GBIF/iNaturalist/EDDMapS where each bbox is a single point. The idea is that you can load up a database of known locations for a tree species and find them in satellite imagery. I think this change may break that use case. 

Why was it necessary? I'm not opposed to considering it, but controversial changes tend to slow down PRs.

Reply:
Because GeoDataset intersections are done at the file level and not feature level, this shouldn't be a problem for point GeoDatasets. Unless there's only one point in a file.

I made this change because I wanted to check that resulting splits didn't intersect, and when joining them I should get the original bbox. At least in my mind two contiguous bboxes aren't intersecting. But we could also solve this by adding a small gap like we did with the time split.

Reply:
GeoDataset intersections are done at the bbox level. Our raster and vector datasets may be at the file level, but these point datasets are at the point level. The entire dataset consists of a single CSV file containing lat/long coords, and each coord is added as a separate element to the R-tree.

I agree that contiguous bboxes shouldn't be intersecting, but that's how rtree defines intersection. In the past, the way we've solved this is for the sampler to ignore patches with 0 area. Does that not work here?

Reply:
Oh I get it, so it wouldn't work in that case. I was removing the change and modifying the non-overlapping test to check for total area == 0 instead of len == 0, but I'm getting a weird R-Tree error. Probably this happened in the first place and that's why I made this change but I forgot.

When attempting this in the tests:
```
# No overlap
assert isclose(_get_total_area(train_ds & val_ds), 0)
assert isclose(_get_total_area(val_ds & test_ds), 0)
assert isclose(_get_total_area(test_ds & train_ds), 0)
```

I get this exception from RTree, only in the third assert:
```
rtree.exceptions.RTreeError: Coordinates must not have minimums more than maximums
```
The first 2 asserts are contiguous bboxes, so the index is not empty but the area is cero. In the third case the index is empty but for some reason the bounds of the index are invalid. Or that is what is seems to me at least.

Reply:
> the index is empty but for some reason the bounds of the index are invalid

This is a known bug in rtree: https://github.com/Toblerity/rtree/issues/204

Does it help to first check if len > 0 and then check if area > 0? Or does that have the same issue?

Reply:
That did it, thanks! I'll rollback this change then.